pkg/dispatcher/dispatcher.go

```diff
@@ -169,13 +169,24 @@ func (dispatcher *Dispatcher) dispatch(ssn *dispatcherframework.Session) {
 
 		// Get the highest priority ResourceBinding from the queue.
 		rbi := resourceBindings.Pop().(*api.ResourceBindingInfo)
-
-		rbi.DispatchStatus = api.UnSuspending
-		dispatcher.cache.UnSuspendResourceBinding(types.NamespacedName{
-			Namespace: rbi.ResourceBinding.Namespace,
-			Name:      rbi.ResourceBinding.Name,
-		})
-		dispatchResourceBindingCount++
+		if ssn.Allocatable(queue, rbi) {
+			rbi.DispatchStatus = api.UnSuspending
```

Comment:
The cache update(Binding) should be after ssn.Allocate, becaue ssn.Allocate may fail, we should only dispatch rb when ssn.Allocate succeed, just like what volcano behaves.